File: coffee-system/src/main/kotlin/com/jesperancinha/coffee/system/CoffeeParadigmsMain.java
package com.jesperancinha.coffee.system;

import jakarta.xml.bind.JAXBException;
import org.kohsuke.args4j.CmdLineException;
import org.kohsuke.args4j.CmdLineParser;

import java.io.FileNotFoundException;

/**
 * Created by joaofilipesabinoesperancinha on 30-04-16.
 */
public class CoffeeParadigmsMain {
    public static void main(String[] args) throws
            CmdLineException,
            FileNotFoundException,
            JAXBException {
        final CoffeParadigmsOptions options = new CoffeParadigmsOptions();
        final CmdLineParser parser = new CmdLineParser(options);
        parser.parseArgument(args);
        options.run();
    }
}

File: jeorg-kotlin-apps/coffee-paradigms/coffee-system/src/main/kotlin/org/jesperancinha/coffee/system/CoffeeParadigmsMain.kt
package org.jesperancinha.coffee.system

import jakarta.xml.bind.JAXBException
import org.springframework.boot.autoconfigure.SpringBootApplication
import picocli.CommandLine
import java.io.FileNotFoundException
import kotlin.system.exitProcess

@SpringBootApplication
open class CoffeeParadigmsMain
/**
 * Created by joaofilipesabinoesperancinha on 30-04-16.
 */
@Throws(FileNotFoundException::class, JAXBException::class)
fun main(args: Array<String>) {
    val options = CoffeeParadigmsOptions()
    val executionResultCode = CommandLine(options).execute(*args)
    exitProcess(executionResultCode)
}

//fun initAll() {
//    preProcessor.initExecutors()
//    coffeeProcessor.initExecutors()
//    paymentProcessor.initExecutors()
//    postProcessor.initExecutors()
//}